documentation/specs/Menu.md

```diff
@@ -0,0 +1,391 @@
+
+```ts
+// Same type definition from @react-aria/Placement
+// prettier-ignore
+type Placement = 'bottom' | 'bottom left' | 'bottom right' | 'bottom start' | 'bottom end' |
+    'top' | 'top left' | 'top right' | 'top start' | 'top end' |
+    'left' | 'left top' | 'left bottom' | 'start' | 'start top' | 'start bottom' |
+    'right' | 'right top' | 'right bottom' | 'end' | 'end top' | 'end bottom';
+
+type Menu = {
+  /** The trigger and menu to render. */
+  children: ReactNode;
+
+  /** Whether the overlay is open by default (uncontrolled). */
+  defaultOpen?: boolean;
+
+  /** Whether the tooltip should be disabled, independent from the trigger. */
+  isDisabled?: boolean;
+
+  /** Whether the overlay is open by default (controlled). */
+  isOpen?: boolean;
+
+  /** Handler that is called when the overlay's open state changes. */
+  onOpenChange?: (isOpen: boolean) => void;
+};
+
+type MenuTrigger = {
+  /** The element that will activate the menu. */
+  children: ReactElement;
+};
+
+type MenuOverlay = {
+  /** The menu sections and items to render. */
+  children: ReactNode;
+
+  /** The item keys that are disabled. These items cannot be selected, focused, or otherwise interacted with. */
+  disabledKeys?: Iterable<Key>;
+
+  /** Item objects in dynamic collections. */
+  items?: Iterable<T>;
+
+  /** Handler that is called when an item is selected. */
+  onAction?: (key: Key) => void;
+
+  /** Handler that is called when the menu should close after selecting an item. */
+  onClose?: () => void;
+
+  /**
+   * The placement of the element with respect to its anchor element.
+   * @default 'top'
+   */
+  placement?: Placement;
+};
+
+// Same subset of type definition from @react-stately/Section
+type MenuSection = {
+  /** An accessibility label for the section. */
+  "aria-label"?: string;
+
+  /** Static child items or a function to render children. */
+  children: ItemElement<T> | ItemElement<T>[] | ItemRenderer<T>;
+
+  /** Item objects in the section. */
+  items?: Iterable<T>;
+};
+
+// Same subset of type definition from @react-stately/Item
+type MenuItem = {
+  /** Rendered contents of the item or child items. */
+  children: ReactNode;
+};
+```
+
+### Example Usage
+
+_Static collection:_
+
+A separator will be rendered between sections.
+
+```tsx
+import { Menu } from "@easypost/easy-ui/Menu";
+
+function Component() {
+  return (
+    <Menu>
+      <Menu.Trigger>
+        <Button>Actions</Button>
+      </Menu.Trigger>
+      <Menu.Menu onAction={(key) => {}}>
+        <Menu.Section>
+          <Menu.Item key="edit">Edit</Menu.Item>
+          <Menu.Item key="duplicate">Duplicate</Menu.Item>
+        </Menu.Section>
+        <Menu.Section>
+          <Menu.Item key="copy">Copy</Menu.Item>
+          <Menu.Item key="cut">Cut</Menu.Item>
+          <Menu.Item key="paste">Paste</Menu.Item>
+        </Menu.Section>
+      </Menu.Menu>
```

Comment:
i updated the component to `Menu` which i like. however, i'm not sure i love `<Menu.Menu>` to signify the part of the component that holds the items just because it looks a little strange to repeat the word. other options could be `<Menu.Items>` or `<Menu.Overlay>` or something else?

Reply:
hmm, id be ok with `<Menu.Items>` as it defines the collection/element relationship with `<Menu.Item>` but also understand that its not ideal either.  i know in an earlier comment I said I prefer `Menu` to `ActionMenu` but id also be fine if you changed it up to `ActionMenu` to avoid the repeated word.

Reply:
checking other libraries, i see these names used for the same component:

`<Menu.List>`, `<Menu.Overlay>`, and `<Menu.Menu>`

any stand out to you / make more sense than the others?

Reply:
I think I have a slight preference for `<Menu.List>` over the others, but i also dont particularly love it either.